My FC work

nathanielhourt opened the pull request:
This is the latest of my FC work, including my merges from BitShares FC and some new work I've done on top to support my ongoing development.

oxarbitrage committed:
add license mit

pmconrad committed:
Document licensing and history

abitmore committed:
Update copyright notice

pmconrad committed:
Updated bloom_filter.hpp from https://github.com/ArashPartow/bloom/blob/c89c2912c445dd5c0afe3d819ca012a6c6639e4e/bloom_filter.hpp

emfrias committed:
Add a fc_[diwe]dump() function, analogous to fc_[diew]log()

heifner committed:
Add exception handling to to_string and to_detail_string

spartucus committed:
Fix non-void function return value

Since last catch insert exception string to `ss`, it should returns.

crypto-ape committed:
Update build system for MinGW

crypto-ape committed:
Updated Boost searching

crypto-ape committed:
Fixed `pthread` dependency

crypto-ape committed:
CMake version bump

crypto-ape committed:
BoostConfig EOF newline

crypto-ape committed:
Updated CMake for MinGW

crypto-ape committed:
Legacy/CMakeLists.txt EOF newline

pmconrad committed:
Dont inline generic pack/unpack to enable externalized serialization

pmconrad committed:
Get rid of some inlines

crypto-ape committed:
Bumped Boost version

crypto-ape committed:
Removed unneeded definitions

crypto-ape committed:
Updated Boost Lockfree compilation test

pmconrad committed:
Separate exception declaration and implementation

oxarbitrage committed:
fix remove appender()

oxarbitrage committed:
fix tests

oxarbitrage committed:
chanmge comment from hostname to IP

oxarbitrage committed:
lambda capture only whats required

pmconrad committed:
Replace one more "new promise" with "promise::create"

pmconrad committed:
Attempt to fix uint128 for mac

pmconrad committed:
Use array::data instead of array::begin

pmconrad committed:
Allow more space for windows sockets due to fc::future changes

crypto-ape committed:
CheckLibcxxAtomic after Boost

nathanielhourt committed:
Improve reflection system and static_variant

Previously, fc reflection generated visitor infrastructure capable of
visiting each field of a struct; however, this is only useful for run-
time operations on structures. No compile-time information was preserved
about the fields of a struct, making compile-time introspection of types
impossible.

Now FC reflection generates rich compile-time information about the
members of structs, including types, names, and structure, allowing
code to be written to explore types in detail at compile-time and fetch
arbitrary fields at runtime without needing to iterate over the unwanted
fields.

To make this a reality, a new `typelist` type was added to store this
compile-time information. This type is also useful within the context of
`static_variant`, as SV previously used implementation detail types in a
private namespace to provide this functionality. Now `static_variant`
uses the reusable functionality of `typelist` to operate, dramatically
reducing the amount of code dedicated to `static_variant` and also
making `static_variant` types more flexible to work with since their
infrastructure is now based on the `typelist` public interface.

abitmore committed:
Update a comment to avoid confusing doxygen

abitmore committed:
Correctly close websocket API connection

jmjatlanta committed:
Merge pull request #142 from bitshares/fix-api-close

Correctly close websocket API connection

oxarbitrage committed:
Merge pull request #141 from bitshares/abitmore-patch-1

Update a comment to avoid confusing doxygen

jmjatlanta committed:
Add hash160 md algo

jmjatlanta committed:
Remove unnecessary predeclarations

jmjatlanta committed:
Fix spacing

jmjatlanta committed:
Add hash160 test

abitmore committed:
Merge pull request #121 from crypto-ape/master_mingw

CMake cleanup & MinGW support

pmconrad committed:
Replaced more instances of array::begin with array::data

pmconrad committed:
Replaced more begin() with data()

pmconrad committed:
Replaced more array::begin() with data()

pmconrad committed:
Removed unused var

jmjatlanta committed:
bump websocketpp for Boost 1.70

pmconrad committed:
Try to work around __int128 problem on mac

pmconrad committed:
Resolved another begin/data inconsistency

pmconrad committed:
Fixed type of chr37

jmjatlanta committed:
Merge pull request #144 from bitshares/jmj_ws_boost170

Bump websocketpp for Boost 1.70

pmconrad committed:
Fix cmake warning

nathanielhourt committed:
Resolve #1898: Fix crash in API

We were getting the moral equivalent of a null pointer dereference by
incrementing an iterator past its past-the-end position, then
dereferencing it. This is now fixed.

jmjatlanta committed:
Merge pull request #140 from pmconrad/1584_more_simplification

Fix simplifications

crypto-ape committed:
support openbsd // make code compliant with clang

crypto-ape committed:
support openbsd // get rid of alloca

crypto-ape committed:
support openbsd // compile with libressl

crypto-ape committed:
support openbsd // implement tcp keep-alive polling

abitmore committed:
Update test case for optional API arguments

Reproduces https://github.com/bitshares/bitshares-core/issues/1898

abitmore committed:
Merge pull request #145 from nathanhourt/bts_1898

Resolves issue https://github.com/bitshares/bitshares-core/issues/1898: Fix crash in API

jmjatlanta committed:
Increase size for Boost 1.70

abitmore committed:
Merge pull request #146 from bitshares/optional-api-fix

Fix optional API arguments

jmjatlanta committed:
Merge pull request #147 from bitshares/jmj_winfix

Increase size for Boost 1.70

jmjatlanta committed:
Remove custom FindBoost.cmake

pmconrad committed:
Merge pull request #148 from crypto-ape/support_building_on_open_bsd

Support Building on OpenBSD

pmconrad committed:
Fix build with older cmake

pmconrad committed:
Fix core #1935

sierra19XX committed:
GRPH-46-AddQuitCommandToCliWallet - Add Quit command to CLI Wallet

pmconrad committed:
Merge pull request #150 from pmconrad/tcp_socket_size

Fix core #1935

jmjatlanta committed:
Merge pull request #149 from bitshares/jmj_cmake

Remove custom FindBoost.cmake

pmconrad committed:
Fix core #1948 - editline does not build

pmconrad committed:
Adjust whitespace

pmconrad committed:
Merge pull request #151 from pmconrad/1948_fix_editline

Fix editline build on non-win32

pmconrad committed:
Possible fix for core #1952

crypto-ape committed:
fix array declaration for broken msvc static expression evaluation

nathanielhourt committed:
Replace improper static_variant operator overloads with comparators

nathanielhourt committed:
Fix warnings

The `appender` class is intended to be inherited and used polymorphically, but does not have a virtual destructor, which causes warnings. This adds a virtual destructor to fix these warnings.

nathanielhourt committed:
Merge pull request #155 from bitshares/nathanhourt-fix-logger-warnings

Fix warnings (ref bitshares/bitshares-core#1584)

nathanielhourt committed:
Fix constexprs, unused code

pmconrad committed:
Merge pull request #153 from crypto-ape/fix_for_msvc_broken_static_evaluation_in_array_declaration

Fix Compilation with MSVC

nathanielhourt committed:
Fixes from code review

pmconrad committed:
Merge pull request #152 from pmconrad/1952_fix_assert

Possible fix for core #1952

nathanielhourt committed:
Merge pull request #154 from nathanhourt/sv_overloads

Replace improper static_variant operator overloads with comparators

nathanielhourt committed:
Add static_variant::operator==

For real this time :)

abitmore committed:
Merge pull request #157 from bitshares/for-core-3-3

Merge for-core-3-3 branch into master

pmconrad committed:
Fix core #314 - Use windows default CAs for tls client connections

nathanielhourt committed:
Merge pull request #156 from nathanhourt/reflection-updates

Improve reflection system and static_variant

nathanielhourt committed:
Fix build on updated G++

nathanielhourt committed:
Merge pull request #158 from nathanhourt/master

Fix build on G++ 7

nathanielhourt committed:
Add FC_REFLECT_DERIVED_NO_TYPENAME

This is moved to FC from graphene/protocol/types.hpp

nathanielhourt committed:
Merge pull request #160 from nathanhourt/master

Add FC_REFLECT_DERIVED_NO_TYPENAME

pmconrad committed:
Merge pull request #159 from pmconrad/314_windows_default_cas

Fix core #314 - Use windows default CAs for tls client connections

pmconrad committed:
Check + rotate logs while logging not by scheduled task

pmconrad committed:
Move log file deletion into separate task

pmconrad committed:
Avoid re-computing seconds

pmconrad committed:
Move helper classes into fc::detail

pmconrad committed:
Moved stuff into fc::test namespace

pmconrad committed:
Delete copy constructor + copy assignment in openssl type wrappers

pmconrad committed:
Delete copy constructor + copy assignment in tcp_socket

pmconrad committed:
Replaced spin_yield_lock with atomics

pmconrad committed:
Print stacktrace when yielding within an exception handler

pmconrad committed:
Preserve crash in debug mode

pmconrad committed:
Add log message

pmconrad committed:
Clean up before returning

pmconrad committed:
Avoid time calculation when not rotating

pmconrad committed:
Merge pull request #166 from pmconrad/copy_safety

Copy safety

pmconrad committed:
Log stacktrace instead of writing to stderr

pmconrad committed:
Merge pull request #162 from pmconrad/1982_rotate_logs

Check + rotate logs while logging not by scheduled task

pmconrad committed:
Fixed assert

pmconrad committed:
Changes to avoid conflict in following cherry-pick

pmconrad committed:
Partial revert "Changes to avoid conflict in following cherry-pick"

This reverts commit d729816b06651ccde50940c92db729cf9fb4c61e.

pmconrad committed:
Whitespace fixes

pmconrad committed:
Removed superfluous return

pmconrad committed:
Merge pull request #167 from pmconrad/exception_format_fix

Exception format fix

pmconrad committed:
Added zero_initialized_array template to restore functionality of fc::array<unsigned char,N>

pmconrad committed:
Merge pull request #164 from pmconrad/1895_namespaces

Move stuff into namespaces

Commit:
fix compilation errors

pmconrad committed:
Reformatting

pmconrad committed:
Added test for zero-initialized arrays

pmconrad committed:
Merge pull request #165 from pmconrad/node_crashes

Changes wrt recent crashes of nodes

pmconrad committed:
Added std::tuple_size for master compatibility

pmconrad committed:
Merge pull request #170 from pmconrad/node_crashes

Make for-core-3-3 branch mergable with master

abitmore committed:
Merge pull request #169 from bitshares/for-core-3-3

Merge 3.3.x branch to master branch

pmconrad committed:
Removed checks for obsolete boost version numbers

pmconrad committed:
Merge pull request #171 from pmconrad/obsolete_boost

Removed checks for obsolete boost version numbers

nathanielhourt committed:
Fix warning in zeroed_array.hpp

Only one template declaration is needed here, I think. Clang gives many, many warnings over it.

pmconrad committed:
Updated license file

pmconrad committed:
Fixed stacktrace license

pmconrad committed:
Get rid of most extra cmake modules

pmconrad committed:
License clarification

nathanielhourt committed:
Merge pull request #175 from bitshares/nathanhourt-fix-warnings

Fix warning in zeroed_array.hpp

sierra19XX committed:
GRPH134-Witness High CPU Issue, websocket changes

bobinson committed:
Merge pull request #7 from satyakoneru/latest-fc

GRPH134-Witness High CPU Issue, websocket changes

pmconrad committed:
Removed unused reflection

pmconrad committed:
Merge pull request #177 from bitshares/cmake_simplification

Get rid of most extra cmake modules

pmconrad committed:
Merge pull request #176 from bitshares/bloom_update

Updated bloom_filter.hpp…

pmconrad committed:
Replaced "regime"

pmconrad committed:
Merge pull request #13 from bitshares/oxarbitrage-patch-2

add license mit

abitmore committed:
Fix potential out-of-bounds access

Thanks to Beosin(https://github.com/Lianantech)

abitmore committed:
Merge pull request #178 from bitshares/abitmore-patch-1

Fix potential out-of-bounds access

srpatel19590 committed:
Fix memory leak. Not all tasks are deleted in thread_d dtor

pmconrad committed:
Added initial workflow

pmconrad committed:
Install boost

pmconrad committed:
Build and run in parallel

pmconrad committed:
Add ccache

pmconrad committed:
Attempt to fix tests

pbattu123 committed:
Merge pull request #8 from peerplays-network/feature/GRPH-147

Fix memory leak. Not all tasks are deleted in thread_d dtor

pmconrad committed:
Added badge to README

pmconrad committed:
Fix for OSX build

oxarbitrage committed:
Merge pull request #181 from pmconrad/2070_osx_fix

Fix for OSX build

pmconrad committed:
Merge pull request #182 from bitshares/for-core-4.0.x

bring macos build fix to master

pmconrad committed:
Fixed badge

pmconrad committed:
Merge pull request #180 from pmconrad/2054_github_autobuild

Add GitHub autobuild

pbattu123 committed:
Merge pull request #9 from peerplays-network/feature/GRPH-46

GRPH-46-AddQuitCommandToCliWallet - Add Quit command to CLI Wallet

pmconrad committed:
Fixed busy-wait in delete_files

pmconrad committed:
Merge pull request #184 from pmconrad/2078_file_appender

Fixed busy-wait in delete_files

pbattu123 committed:
Merge branch 'latest-fc' into feature/GRPH-92

pbattu123 committed:
Merge pull request #13 from gladcow/feature/GRPH-92

Build optimization fc changes: GRPH-92

pbattu123 committed:
Revert "Merge pull request #13 from gladcow/feature/GRPH-92"

This reverts commit c30f3b7d32502fc1a746cefa132848cb80a4129b, reversing
changes made to 31e289c53d3625afea87c54edb6d97c3bca4c626.

pbattu123 committed:
include fc build optimization changes

abitmore committed:
Merge pull request #185 from bitshares/for-core-4.0.x

Merge 4.0.x changes to master

abitmore committed:
Return a user-friendly msg on RPC execution error

abitmore committed:
Fix Github auto-build

Remove the predefined boost related environment variables.

abitmore committed:
Merge pull request #187 from bitshares/fix-github-autobuild

Fix Github auto-build

abitmore committed:
Merge pull request #186 from bitshares/abitmore-patch-1

Return a user-friendly message on RPC execution error

abitmore committed:
Merge pull request #188 from bitshares/for-core-4.0.x

Merge 4.0.x changes to master

nathanielhourt committed:
Fix Build

This is necessary to build in my environment for both clang and g++

Clang 9.0.1
G++ 9.2.1
Boost 1.72.0

nathanielhourt committed:
Fix Build

Definitely something strange going on here, as in my environment
the value of Boost_UNIT_TEST_FRAMEWORK_LIBRARY is
"Boost::unit_test_framework"... So this is my fix, but probably
something else should be done.

Boost 1.72.0

abitmore committed:
Merge pull request #189 from nathanhourt/fix-1

Fix Build

abitmore committed:
Merge pull request #190 from nathanhourt/fix-2

Fix Build

abitmore committed:
Update bool serialization

Make the logic clearer, to avoid issues on some compilers.
Thanks to @pureland and team.

abitmore committed:
Merge pull request #192 from bitshares/update-bool-pack

Update bool serialization

abitmore committed:
Fix potential infinity loop

Fix exception::to_detail_string(...)

abitmore committed:
Update to_variant(size_t) for macOS and OpenBSD

Fix to_variant(size_t) for OpenBSD, move implementation for macOS to cpp file

abitmore committed:
Replace int with 64-bit types in static_variant

abitmore committed:
Merge pull request #193 from bitshares/fix-exception-to-string

Fix potential infinity loop in `exception::to_detail_string()`

abitmore committed:
Merge pull request #195 from bitshares/replace-int-in-static-variant

Replace int with 64-bit types in static_variant

abitmore committed:
Merge pull request #194 from bitshares/update-variant-size-t-macos-openbsd

Update to_variant(size_t) for macOS and OpenBSD

abitmore committed:
Fix compiler warnings about static_variant

abitmore committed:
Merge pull request #196 from bitshares/fix-static-variant-warning

Fix compiler warnings about static_variant

jmjatlanta committed:
Merge remote-tracking branch 'upstream/for-core-4.0.x' into jmj_bsip64

jmjatlanta committed:
improve hash160 performance

jmjatlanta committed:
define hash160 reset method

jmjatlanta committed:
add impl to hash160

abitmore committed:
Merge pull request #143 from bitshares/jmj_bsip64

Add Hash160 algorithm

abitmore committed:
Update run-parallel-tests.sh to support arguments

abitmore committed:
Show more msgs when testing with Github Actions

abitmore committed:
Show more messages in Travis CI

abitmore committed:
Merge branch 'master' into for-core-4.0.x

abitmore committed:
Merge pull request #197 from bitshares/improve-testing

Update run-parallel-tests.sh to support arguments

abitmore committed:
Fix static_variant_depth_test

abitmore committed:
Merge pull request #199 from bitshares/abitmore-patch-1

Fix static_variant_depth_test

abitmore committed:
Merge pull request #200 from bitshares/merge-core-4.0.x-to-master

Merge changes done for bitshares-core 4.0 to master

abitmore committed:
Refactor code about logging remote host info

abitmore committed:
Fix test cases

abitmore committed:
Tweak code slightly

abitmore committed:
Merge 'master' branch

abitmore committed:
Revert to simple WS connection for clients

abitmore committed:
Add todo

abitmore committed:
Rename a function to avoid accidental misuse

abitmore committed:
Remove unused header inclusion

abitmore committed:
Update RPC logging format

abitmore committed:
Update test cases about RPC logging level

abitmore committed:
Add test cases about on_http for websocket servers

abitmore committed:
Remove unused websocket_tls_client class

The funcionality is already covered by the websocket_client class

abitmore committed:
Fix websocket_client::secure_connect()

Code synced from connect(), added todo to refactor later

abitmore committed:
Merge pull request #134 from bitshares/jmj_844b

Log host information from websocket requests

abitmore committed:
Refactor websocket_client, fix duplicate code

Fix duplicate code issue in websocket_client::connect() and
websocket_client::secure_connect(), and other minor issues

abitmore committed:
Remove unused types, wrap long lines

abitmore committed:
Refactor websocket_tls_server_impl with template

Removes unnecessary differences between websocket_server_impl and websocket_tls_server_impl.

abitmore committed:
Capture only required vars, update coding style

and replace an assertion with a check.

abitmore committed:
Add notes

abitmore committed:
Refactor destructor of websocket_server_impl

abitmore committed:
Fix Clang -Wmismatched-tags warning on tuple_size

abitmore committed:
Merge pull request #202 from bitshares/fix-tuple-size-warning

Fix Clang -Wmismatched-tags warning on tuple_size

abitmore committed:
Update coding style

abitmore committed:
Add promise to wait for closure of server socket

This may fix the issues that a websocket server may crash or hang when quitting

abitmore committed:
Add missing functions to websocket_tls_server

These functions were in websocket_server class but not in websocket_tls_server class:
- get_listening_port
- stop_listening
- close

abitmore committed:
Merge pull request #201 from bitshares/ws-refactory

Websocket code refactory and bugfix

abitmore committed:
Merge pull request #204 from bitshares/jmj_1256

Test to demonstrate fc::mutex and fc::thread

abitmore committed:
Fix race condition in websocket server close

Note: the close() functions are not yet used in bitshares-core, but only in tests in FC.
The new code is not fully thread-safe either.

abitmore committed:
Merge pull request #205 from bitshares/fix-websocket-close

Fix race condition in websocket server close

abitmore committed:
Add macOS to Github Autobuild

abitmore committed:
Fix paths for macOS unit tests in github autobuild

abitmore committed:
Update time_test to get around a boost 1.60 issue

abitmore committed:
Update boost version checks in time_test to macros

abitmore committed:
Merge pull request #206 from bitshares/github-auto-build

Add macOS to Github Autobuild

pravblockc committed:
websocket fix, issue 13

pravblockc committed:
Revert "websocket fix, issue 13"

This reverts commit 0cf6f461b682e936dac05b882bdd5b7799f3afe3

pravblockc committed:
websocket issue

roshsyed committed:
Added .gitlab-ci.yml

bobinson committed:
Merge branch 'issue-13-fix-websocket' into 'latest-fc'

Issue-13-fix (websocket shutdown)

See merge request PBSA/PeerplaysIO/tools-libs/peerplays-fc!17

pravblockc committed:
ws-updates

bobinson committed:
Merge branch 'ws-fc-updates' into 'latest-fc'

ws-updates

See merge request PBSA/PeerplaysIO/tools-libs/peerplays-fc!18

nathanielhourt committed:
Add hash_ctr_rng.hpp

This file was originally added in ff099209b6510412e45ac83fd67f5b8e9a2ec1ea

nathanielhourt committed:
Fix warnings

nathanielhourt committed:
websocket_server default empty XFF header

For some reason, the websocket_server types have been altered to require
an X-Forwarded-For header (???) which seems utterly unnecessary. AFAICT,
it's safe to just leave it empty (BitShares does), so add a default arg
to make it do so by default.

nathanielhourt committed:
Link boost.serialization

nathanielhourt committed:
Revert "Removed signed_int"

This reverts commit 1dcacbafc9e85552e8bd9b8b5082e1ce3364fe43.

nathanielhourt committed:
Use signed_int to serialize enums

pravblockc committed:
static variant update

nathanielhourt committed:
Add option to compile dynamic libraries

Also, fix build:
Clang 10.0.1 has incompatible rules with other compilers on whether
you must/cannot put the `template` keyword in part of the reflection
code. Add a special case to make it work all around.

pravblockc committed:
Feature/readline-editline

bobinson committed:
Merge branch 'feature/rl-editline' into 'latest-fc'

Feature/readline-editline

See merge request PBSA/PeerplaysIO/tools-libs/peerplays-fc!20

bobinson committed:
Merge branch 'feature/static_variant_update' into 'latest-fc'

static variant update

See merge request PBSA/PeerplaysIO/tools-libs/peerplays-fc!19

Commit:
Remove as much warnings from build log as possible

nathanielhourt committed:
Merge remote-tracking branch 'ppu/latest-fc' into bts-updates

nathanielhourt committed:
Restore pre-merge websocketpp

This is just a rote copy of vendor/websocketpp from two commits
ago, ie 0e9259486cb04a90ec709750143cc5be1da474ff

serkixenos committed:
Merge pull request #17 from peerplays-network/fc/cleaner-build-log

Remove as much warnings from build log as possible

bobinson commented:
@nathanhourt  - can you send the same to gitlab so that the CI gets kicked in ? We need to merge this as there is more sidechain work just started and want to use this FC changes. Just send the MR in gitlab and I guess we will find a way to quickly merge it

nathanielhourt committed:
Compiler incompatibility

It broke again so poke it again. Perhaps all clang versions require `template` here? I'm not sure, haven't gotten around to testing all the different compilers on it yet.

MichelSantos committed:
Merge branch latest-fc into dapp-support

nathanielhourt committed:
Merge pull request #1 from MichelSantos/dapp-support

Merge with latest-fc

nathanielhourt committed:
Fix build: adjust tcp_socket allocation
